### ISSUE TITLE
Addon-docs: Source editing support

### ISSUE THREAD

**Issue by shilman** [storybooks/storybook#6642]
Add editing support to #6641

Details TBD

**Comment by libetl** [comment]
What do we want here ?

**Comment by libetl** [comment]
I can try to help

**Comment by shilman** [comment]
@libetl Haven't really spec'ed it out yet. Will definitely loop you in once this gets started.

There are some "lightweight" in-place code editors like https://github.com/FormidableLabs/react-live and I'd like to support this interaction. However, I'm unclear how this will integrate with the rest of the Storybook UI and also cross-framework.

More soon cc @elevatebart

**Comment by kevinSuttle** [comment]
I've seen how the folks at Segment handle this: 

https://github.com/segmentio/evergreen/blob/7fc8135c2b80afd6c7158853f7ffdf4acfffbced/docs/src/components/Playground.js

https://github.com/segmentio/evergreen/blob/7fc8135c2b80afd6c7158853f7ffdf4acfffbced/docs/src/components/MDXPlayground.js

Live example: https://evergreen.segment.com/components/button/

**Comment by kevinSuttle** [comment]
Is https://github.com/storybookjs/storybook/releases/tag/v5.2.0-beta.8 related to this?

**Comment by shilman** [comment]
@kevinSuttle thanks for the link. this is probably a 5.3 feature and still unclear how to fit it in in a cross-framework way!

**Comment by kevinSuttle** [comment]
> still unclear how to fit it in in a cross-framework way!

Definitely. That's not an easy task for sure. Thanks for your work on this!

**Comment by elevatebart** [comment]
For now, the [`vue-inbrowser-compiler`](https://www.npmjs.com/package/vue-inbrowser-compiler) for VueJs. 
We might leverage this but it means definitely a lot of work to get cross framework compatibility.

**Comment by bodograumann** [comment]
react-styleguidist / vue-styleguidist allow this. We are looking to use addon-docs instead of styleguidist. Would be great if editing was allowed. Maybe have a look at their solution.

**Comment by shilman** [comment]
@bodograumann we have the `vue-styleguidist` maintainer, @elevatebart, on our team now -- if anybody can figure it out, it's him!

**Comment by jonsalvas** [comment]
any updates on this? Really useful feature imo

**Comment by valentinpalkovic** [comment]
Source editing in docs never shipped, and the underlying `@storybook/addon-storysource` / `@storybook/source-loader` it depended on were both removed in Storybook 9.0 (replaced by a read-only Code Panel in `@storybook/addon-docs`). Storybook's direction is read-only source surfacing, not in-page editing. Closing as won't-do — if you need an editable playground, a custom addon or external playground (CodeSandbox/StackBlitz) is the right path.
